internal/datastore/common/logging.go

```diff
@@ -0,0 +1,15 @@
+package common
+
+import (
+	"context"
+
+	log "github.com/authzed/spicedb/internal/logging"
+)
+
+// LogOnError executes the function and logs the error.
+// Useful to avoid silently ignoring errors in defer statements
+func LogOnError(ctx context.Context, f func() error) {
+	if err := f(); err != nil {
+		log.Ctx(ctx).Error().Err(err).Msg("datastore error")
```

Comment:
`Error()` here is redundant. From zerolog API:

```go
// Err starts a new message with error level with err as a field if not nil or
// with info level if err is nil.
//
// You must call Msg on the returned event in order to send the event.
func (l *Logger) Err(err error) *Event {
	if err != nil {
		return l.Error().Err(err)
	}

	return l.Info()
}
```

Reply:
That's what I get for copy-pasting :)

Reply:
Fixed